### PR TITLE
Prevent setting of nextDom on textnodes

### PR DESCRIPTION
**Pull request by JoviDeCroock**
Fixes: https://github.com/preactjs/preact/issues/2455

### CODE REVIEW

File: src/diff/index.js [hunk 1]
```diff
@@ -439,7 +439,9 @@ export function unmount(vnode, parentVNode, skipRemove) {
 
 	// Must be set to `undefined` to properly clean up `_nextDom`
 	// for which `null` is a valid value. See comment in `create-element.js`
-	vnode._dom = vnode._nextDom = undefined;
+	if (vnode.type) {
```
Comment (JoviDeCroock) [inline src/diff/index.js:442]: Drafted this since I want to see if there's a better solution

**Comment by developit** [comment]
The interesting thing is that we should never have an un-normalized VNode in `unmount()`. Somehow we're getting a String, which means somewhere we're setting `._children = .props.children` or something...

**Comment by JoviDeCroock** [comment]
> The interesting thing is that we should never have an un-normalized VNode in `unmount()`. Somehow we're getting a String, which means somewhere we're setting `._children = .props.children` or something...

Yes, due to `dangerouslySetInnerHTML` I guess

**Comment by developit** [comment]
ah! yeah so this line actually sets `vnode._children` to the wrong value (un-normalized):
https://github.com/preactjs/preact/blob/master/src/diff/index.js#L365

in all cases _except_ dangerouslySetInnerHTML, we end up resetting it here back to the correct value:
https://github.com/preactjs/preact/blob/master/src/diff/children.js#L60

**Comment by JoviDeCroock** [comment]
Do we want to make `dangerouslySetInnerHTML` a vnode? or does this low size fix work for you?

**Comment by developit** [comment]
Alright I think I found the spot. We are intentionally overloading `_children` as "temporary new children" any time we jump into `diffChildren` - the issue here is that we aren't calling diffChildren, so it never gets renormalized.

https://github.com/preactjs/preact/pull/2458

File: src/diff/index.js [hunk 2]
```diff
@@ -439,7 +441,9 @@ export function unmount(vnode, parentVNode, skipRemove) {
...
+		vnode._dom = vnode._nextDom = undefined;
+	}
```
Comment (developit) [inline src/diff/index.js:446]:
```suggestion
	vnode._dom = vnode._nextDom = undefined;
```

**Comment by developit** [comment]
Testing now with only the normalization change and no patch change.

**Pull request merged**